frontend/src/metabase/lib/formatting/link.ts

```diff
@@ -0,0 +1,100 @@
+import { isDate } from "metabase/lib/schema_metadata";
+
+import { formatValue } from "metabase/lib/formatting";
+import { formatDateTimeForParameter } from "./date";
+
+import type { DatasetColumn } from "metabase-types/api/dataset";
+
+interface TemplateForClickFormatFunctionParamsType {
+  value: string;
+  column: DatasetColumn;
+}
+
+export interface ValueAndColumnForColumnNameDate {
+  column: DatasetColumn;
+  parameterBySlug: string;
+  parameterByName: string;
+  userAttribute: string;
+}
+
+function formatValueForLinkTemplate(value: string, column: DatasetColumn) {
+  if (isDate(column) && column.unit) {
+    return formatDateTimeForParameter(value, column.unit);
+  }
+  return value;
+}
+
+export function renderLinkTextForClick(
+  template: string,
+  data: ValueAndColumnForColumnNameDate,
+) {
+  return renderTemplateForClick(
+    template,
+    data,
+    ({ value, column }: { value: string; column: DatasetColumn }) =>
```

Comment:
nit: Use the same type as line 46 😉 
```suggestion
    ({ value, column }: TemplateForClickFormatFunctionParamsType) =>
```

Reply:
Thank you @WiNloSt  👍